### ISSUE TITLE
Editing environment broken

### ISSUE THREAD

**Issue by headtr1ck** [pytest-dev/pytest-html#506]
I was trying to edit the environment information in the html report following the guide, i.e. in conftest.py:
```python
def pytest_configure(config):
    config._metadata["foo"] = "bar"
```
and I get the error:
> INTERNALERROR> AttributeError: 'Config' object has no attribute '_metadata'

Same with `pytest_sessionfinish`.

Using pytest v6 it was working, was there a breaking change (might be also related to pytest-metadata v2)?

I am using
python==3.9.10
pytest==7.1.1
pytest-html==3.1.1
pytest-metadata==2.0.0

**Comment by Northxw** [comment]
I had the same problem as you, I tried looking for the _metadata attribute in the config object in the latest version, but couldn't find it.
then, I checked the chagelog list and found no entries for the relevant changes, expecting an answer.

**Comment by BeyondEvil** [comment]
This was an issue with `pytest-metadata`, should be fixed in `2.0.1`.

**Comment by headtr1ck** [comment]
Thanks!
Updating to `pytest-metadata==2.0.1` solved the problem :)

**Comment by hugobuddel** [comment]
It seems this functionality has been removed again in `pytest-metadata` 3, but downgrading to 2.0.4 worked

**Comment by BeyondEvil** [comment]
> It seems this functionality has been removed again in `pytest-metadata` 3, but downgrading to 2.0.4 worked

Mind sharing some more information?

**Comment by hugobuddel** [comment]
> > It seems this functionality has been removed again in `pytest-metadata` 3, but downgrading to 2.0.4 worked
> 
> Mind sharing some more information?

I have a very similar setup as @headtr1ck describes, and I get the exact same error message. But I already was using pytest-metadata 3.1.1. So I first downgraded to 3.1.0, which didn't help. Then I tried 2.0.4 and everything seems to work.

My thought process was that this was then probably an intentional change, because it was a major version upgrade. And in general accessing attributes that start with an underscore is considered bad form, so having that break in a major version change seemed 'normal' to me. I was mainly commenting here to help other people that ran in the same problem.

If this is supposed to work, then I could maybe try to figure out what exactly is happening, and perhaps create a minimal reproducible example. The code base I'm working on is new to me, as in trying to get it to work for the first time today, so I don't fully understand what is going on.
